source/simulator/helper_functions.cc

```diff
@@ -2403,6 +2404,235 @@ namespace aspect
+                             update_quadrature_points |
+                             update_JxW_values);
+
+    FEValues<dim> fe_values_adjoint (*mapping,
+                                     finite_element,
+                                     quadrature_formula,
+                                     update_values |
+                                     update_gradients |
+                                     update_quadrature_points |
+                                     update_JxW_values);
+
+    FEFaceValues<dim> fe_face_values (*mapping,
+                                      finite_element,
+                                      quadrature_formula_face,
+                                      update_values |
+                                      update_gradients |
+                                      update_quadrature_points |
+                                      update_JxW_values);
+
+    // Storage for shape function values for the current solution.
+    // Used for constructing the known side of the CBF system.
+    std::vector<double> phi_eta (dofs_per_cell);
+    std::vector<double> phi_rho (dofs_per_cell);
+
+    // Initializing local rhs and mass matrix
+    Vector<double> local_rhs(dofs_per_cell);
+    FullMatrix<double> local_mass_matrix(dofs_per_cell, dofs_per_cell);
+
+    // Loop over all of the surface cells and if one less than h/3 away from
+    // one of the top or bottom boundaries, assemble CBF system for it.
+    typename DoFHandler<dim>::active_cell_iterator
+    cell = dof_handler.begin_active(),
+    endc = dof_handler.end();
+
+    std::vector<types::global_dof_index> local_dof_indices (dofs_per_cell);
+
+    // Get a pointer to the dynamic topography postprocessor.
+    const Postprocess::DynamicTopography<dim> &dynamic_topography =
+      postprocess_manager.template get_matching_postprocessor<Postprocess::DynamicTopography<dim> >();
+
+    // Get the already-computed dynamic topography solution.
+    const LinearAlgebra::BlockVector topo_vector = dynamic_topography.topography_vector();
+    std::vector<double> topo_values( quadrature_formula.size() );
+
+    // TODO: get this from dynamic topography postprocessor
+    const double density_above = 0;
+
+    for (; cell!=endc; ++cell)
+      if (cell->is_locally_owned())
+        {
+          fe_values.reinit (cell);
+          fe_values_adjoint.reinit (cell);
+
+          // Evaluate the material properties and the solution within the cell
+          MaterialModel::MaterialModelInputs<dim> in(fe_values, cell, introspection, solution);
+          MaterialModel::MaterialModelOutputs<dim> out(fe_values.n_quadrature_points, introspection.n_compositional_fields);
+          material_model->evaluate(in, out);
+
+          MaterialModel::MaterialModelInputs<dim> in_adjoint(fe_values, cell, introspection, current_adjoint_solution);
+
+          local_rhs = 0.;
+          local_mass_matrix = 0.;
+
+
+          // VOLUME TERM
```

Comment:
Capital letters seem to shout at me while reading the source code ;-). Also remove at least one of the empty lines, it is not that important that a new block that starts here.

```suggestion
          // Assemble the volume term
```